src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,257 @@
+        assertThrows(RegistrationException.class, () -> registrationService.register(user1));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user2));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user3));
+    }
+
+    @Test
+    void register_loginLengthAboveMin_ok() throws RegistrationException {
+        User user1 = new User();
+        User user2 = new User();
+        User user3 = new User();
+        user1.setLogin(GOOD_LOGIN_FIFTH);
+        user2.setLogin(GOOD_LOGIN_SIXTH);
+        user3.setLogin(GOOD_LOGIN_SEVENTH);
+        user1.setPassword(GOOD_PASSWORD_FIRST);
+        user2.setPassword(GOOD_PASSWORD_SECOND);
+        user3.setPassword(GOOD_PASSWORD_THIRD);
+        user1.setAge(AGE_OF_EIGHTEEN);
+        user2.setAge(AGE_OF_TWENTY);
+        user3.setAge(AGE_OF_FORTY);
+        registrationService.register(user1);
+        registrationService.register(user2);
+        registrationService.register(user3);
+        assertEquals(GOOD_LOGIN_FIFTH, user1.getLogin());
+        assertEquals(GOOD_LOGIN_SIXTH, user2.getLogin());
+        assertEquals(GOOD_LOGIN_SEVENTH, user3.getLogin());
+    }
+
+    @Test
+    void register_loginLengthBelowMin_notOk() {
+        User user1 = new User();
+        User user2 = new User();
+        User user3 = new User();
+        user1.setLogin(EMPTY_LOGIN_OR_PASSWORD);
+        user2.setLogin(LOGIN_LENGTH_OF_TWO);
+        user3.setLogin(LOGIN_LENGTH_OF_FOUR);
+        user1.setPassword(GOOD_PASSWORD_FIRST);
+        user2.setPassword(GOOD_PASSWORD_SECOND);
+        user3.setPassword(GOOD_PASSWORD_THIRD);
+        user1.setAge(AGE_OF_EIGHTEEN);
+        user2.setAge(AGE_OF_TWENTY);
+        user3.setAge(AGE_OF_FORTY);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user1));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user2));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user3));
+    }
+
+    @Test
+    void register_passwordLengthAboveMin_ok() throws RegistrationException {
+        User user1 = new User();
+        User user2 = new User();
+        User user3 = new User();
+        user1.setLogin(GOOD_LOGIN_EIGHT);
+        user2.setLogin(GOOD_LOGIN_NINTH);
+        user3.setLogin(GOOD_LOGIN_TENTH);
+        user1.setPassword(GOOD_PASSWORD_FIRST);
+        user2.setPassword(GOOD_PASSWORD_SECOND);
+        user3.setPassword(GOOD_PASSWORD_THIRD);
+        user1.setAge(AGE_OF_EIGHTEEN);
+        user2.setAge(AGE_OF_TWENTY);
+        user3.setAge(AGE_OF_FORTY);
+        registrationService.register(user1);
+        registrationService.register(user2);
+        registrationService.register(user3);
+        assertEquals(GOOD_PASSWORD_FIRST, user1.getPassword());
+        assertEquals(GOOD_PASSWORD_SECOND, user2.getPassword());
+        assertEquals(GOOD_PASSWORD_THIRD, user3.getPassword());
```

Comment:
The `register_passwordLengthAboveMin_ok` test should include a verification step to check if the user is actually added to the storage after registration.